specification/communication/data-plane/Microsoft.CommunicationServicesChat/preview/2020-11-01-preview3/communicationserviceschat.json

```diff
@@ -0,0 +1,1675 @@
+          "type": "string",
+          "readOnly": true,
+          "example": "123456789"
+        }
+      }
+    },
+    "ChatMessage": {
+      "description": "Chat message.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "The id of the chat message. This id is server generated.",
+          "type": "string",
+          "readOnly": true,
+          "example": "123456789"
+        },
+        "type": {
+          "description": "Type of the chat message.\r\n            \r\nPossible values:\r\n    - Text\r\n    - ThreadActivity/TopicUpdate\r\n    - ThreadActivity/AddMember\r\n    - ThreadActivity/DeleteMember",
+          "type": "string",
+          "example": "Text"
+        },
+        "priority": {
+          "$ref": "#/definitions/ChatMessagePriority"
+        },
+        "version": {
+          "description": "Version of the chat message.",
+          "type": "string",
+          "readOnly": true
+        },
+        "content": {
+          "description": "Content of the chat message.",
+          "type": "string",
+          "example": "Come one guys, lets go for lunch together."
+        },
+        "senderDisplayName": {
+          "description": "The display name of the chat message sender. This property is used to populate sender name for push notifications.",
+          "type": "string",
+          "example": "Jane"
+        },
+        "createdOn": {
+          "format": "date-time",
+          "description": "The timestamp when the chat message arrived at the server. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "readOnly": true,
+          "example": "2020-10-30T10:50:50Z"
+        },
+        "senderId": {
+          "description": "The id of the chat message sender.",
+          "type": "string",
+          "readOnly": true,
+          "example": "8:acs:8540c0de-899f-5cce-acb5-3ec493af3800_0e59221d-0c1d-46ae-9544-c963ce56c10b"
+        },
+        "deletedOn": {
+          "format": "date-time",
+          "description": "The timestamp (if applicable) when the message was deleted. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2020-10-30T10:50:50Z"
+        },
+        "editedOn": {
+          "format": "date-time",
+          "description": "The last timestamp (if applicable) when the message was edited. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2020-10-30T10:50:50Z"
+        }
+      }
+    },
+    "ChatMessagesCollection": {
+      "description": "Collection of chat messages for a particular chat thread.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of chat messages.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ChatMessage"
+          },
+          "readOnly": true
```

Comment:
Instead of `readOnly` we should mark properties add properties to `required`. Otherwise AutoREST generates poorly usable code. Properties that are not required, can then be marked readonly where applicable

Reply:
value for example should be required here